sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzurePath.java

```diff
@@ -3,30 +3,80 @@
+ * of the first element in the path. The root component is used to identify which container a path belongs to.
+ * <p>
+ * Constructing a syntactically valid path does not ensure a resource exists at the given path. An error will
+ * not be thrown until trying to access an invalid resource, e.g. trying to access a resource that does not exist.
+ * <p>
+ * Path names are case sensitive.
+ * <p>
+ * If a resource is accessed via a relative path, it will be resolved against the default directory of the file system.
+ * The default directory is as defined in the {@link AzureFileSystem} docs.
+ * <p>
+ * Leading and trailing separators will be stripped. This has the effect of making "foo/" and "foo" equivalent paths.
+ *
  * {@inheritDoc}
  */
 public final class AzurePath implements Path {
+    private final ClientLogger logger = new ClientLogger(AzurePath.class);
+    private static final String ROOT_DIR_SUFFIX = ":";
+
     private final AzureFileSystem parentFileSystem;
     private final String pathString;
 
-    AzurePath(AzureFileSystem parentFileSystem, String s, String... strings) {
+    AzurePath(AzureFileSystem parentFileSystem, String first, String... more) {
+        if (more == null) {
+            more = new String[0]; // Just to make processing easier later. This wont affect the result.
+        }
         this.parentFileSystem = parentFileSystem;
-        this.pathString = String.join(this.parentFileSystem.getSeparator(),
-                Flux.just(s).concatWith(Flux.just(strings)).toIterable());
+        Flux<String> elementFlux =
+            // Strip any trailing, leading, or internal delimiters so there are no duplicates when we join.
+            Flux.fromArray(first.split(this.parentFileSystem.getSeparator()))
+                .concatWith(Flux.fromArray(more)
+                    .flatMap(str -> Flux.fromArray(str.split(this.parentFileSystem.getSeparator()))))
+                .filter(str -> !str.isEmpty());
+
+        this.pathString = String.join(this.parentFileSystem.getSeparator(), elementFlux.toIterable());
+
+        // No element but the first may contain ":"
+        elementFlux.skip(1)
+            .flatMap(str -> str.contains(ROOT_DIR_SUFFIX)
+                ? Mono.error(Utility.logError(logger, new InvalidPathException(this.pathString, ROOT_DIR_SUFFIX
+                + " is an invalid character except to identify the root element of this path if there is one.")))
+            : Mono.just(str)).blockLast();
+
+        // There may only be at most one instance of ":" in the root component, and it must be the last character.
+        elementFlux.take(1)
+            .flatMap(str -> str.contains(ROOT_DIR_SUFFIX) && str.indexOf(ROOT_DIR_SUFFIX) < str.length() - 1
+            ? Mono.error(Utility.logError(logger, new InvalidPathException(this.pathString, ROOT_DIR_SUFFIX + " may"
+                + " only be used as the last character in the root component of a path")))
+            : Mono.just(str)).blockLast();
```

Comment:
Basically I'm mimicking what windows does with its drives/root directories--using <driveName>: to double as the name of the root directory. The ':' helps identify that an element is meant to be the root instead of just any old path element. The offshoot as that ':' can only appear in the first element, otherwise it loses its value in distinguishing the root. If it does appear in the first element, it can only appear as the last character in that element. Does that answer your question? The `FileSystem.getPath()` test cases may also help clarify this. I can clean up the inline docs, too, if you think that would help clarify the logic